### PR TITLE
Upgrade test: increase retry times and add more check

### PR DESCRIPTION
**Pull request by yusuoh**
Fix the flakiness of some e2e tests:
(1) Increase the deadline for deleting namespace. Some tests take more than 120s to delete all jobs and it caused the test cluster not to be completely clean after the test exits. It affects all the cluster-wide tests, where the subsequent tests are sharing the same namespace with previous ones.
(2) Restore a deleted script which is being used by mixer validator e2e test when the flag with_mixer_validation is set.
(3) Add more routing check in upgrade tests. Increase the retry times and merge some duplicate code.

### CODE REVIEW

**Comment by hklai** [comment]
@andraxylia, @costinm please review and approve.

The upgrade tests are consistently failing in release tests which blocks daily releases.

/cc @rkpagadala

**Comment by ldemailly** [comment]
/lgtm

ps: we need to test upgrades with continuous traffic (use `fortio load -qps 5 -t 0 ...` for instance) and ensure there are no errors during the upgrade

(given we do have 503s errors during regular rule change, it's unlikely/aspirational to pass)

File: install/kubernetes/webhook-create-signed-cert.sh [hunk 1]
```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+set -e
+
+usage() {
+    cat <<EOF
+Generate certificate suitable for use with an Istio webhook service.
```
Comment (ldemailly) [inline install/kubernetes/webhook-create-signed-cert.sh]: pretty sure I saw this code elsewhere before... and/or we have stuff checked in that makes it no needed?
@ajy @wattli

File: tests/e2e/tests/upgrade/upgrade_test.go [hunk 2]
```diff
@@ -223,30 +232,13 @@ func applyRules(ruleKeys []string) error {
 	return nil
 }
 
-func checkTraffic(t *testing.T) {
```
Comment (ldemailly) [inline tests/e2e/tests/upgrade/upgrade_test.go]: why is this gone. what is the replacement ?

**Comment by ldemailly** [comment]
can you put more details in the PR description please

**Comment by ldemailly** [comment]
feel free to merge to unblock the release pipeline but I have some questions that can use answers (and/or follow up PR)

File: install/kubernetes/webhook-create-signed-cert.sh (hunk 1 above)
Comment (yusuoh) [reply install/kubernetes/webhook-create-signed-cert.sh]: We no longer need this script for injector webhook so I removed it in #4612 . However, there is an e2e test for admin webhook for mixer validator. So I have to restore it temporarily to fix the tests.

https://github.com/istio/istio/blob/master/tests/e2e/framework/kubernetes.go#L196

File: tests/e2e/tests/upgrade/upgrade_test.go (hunk 2 above)
Comment (yusuoh) [reply tests/e2e/tests/upgrade/upgrade_test.go]: It has been merged into probeGateway as there were some duplicate logic there.

**Comment by yusuoh** [comment]
Added more description. Please merge this PR to fix some flaky e2e tests.

**Comment by linsun** [comment]
@yusuoh can we merge this now?  Seems all tests are passing except the hold label.

**Pull request merged**